chargeOclock/Source/Views/ClientTableViewCell.swift

```diff
@@ -13,29 +13,51 @@ class ClientTableViewCell: UITableViewCell, ClientSettable {
 	@IBOutlet private var nameTextField: UITextField!
 	@IBOutlet private var detailLabel: UILabel!
 
-	private var clientTextFieldDelegate = ClientTextFieldDelegate()
-
-	var updateClient: ((Client) -> Void)?
-
 	var client: Client? = nil {
 		didSet {
 			nameTextField.text = client?.name
 			detailLabel.text = "\(client?.id ?? 0)"
-			nameTextField.delegate = clientTextFieldDelegate
-			clientTextFieldDelegate.endEditingWithText = handleNewClientName
+			nameTextField.delegate = self
 		}
 	}
 
 	override func prepareForReuse() {
 		nameTextField.text = ""
 		detailLabel.text = ""
 	}
+}
+
+extension ClientTableViewCell: UITextFieldDelegate {
 
-	func handleNewClientName(name: String) {
-		client?.name = name
-		guard let client = client else {
+	func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+		textField.resignFirstResponder()
+		return true
+	}
+
+	func textFieldDidEndEditing(_ textField: UITextField) {
+		guard let text = textField.text, let client = client else {
 			return
 		}
-		updateClient?(client)
+		client.name = text
+		client.save { error in
+			guard error == nil else {
+				let alert = UIAlertController(title: "Save error", message: "Client saving failed, error \(error?.localizedDescription)", preferredStyle: .alert)
+				alert.addAction(UIAlertAction(title: "Ok", style: .default, handler: nil ))
+
+				var parentViewController: UIViewController? {
```

Comment:
while this will work, it doesn't really make sense to have this as a calculated property, so just make it into a variable and perform the same operations on it

edit: ah, sorry... after looking through the code in more detail, it is needed as is... ignore this comment 😇